runtime/Dart/lib/src/parser_rule_context.dart

```diff
@@ -38,18 +38,18 @@ class ParserRuleContext extends RuleContext {
   ///  with this rule's context. This is empty for parsing w/o tree constr.
   ///  operation because we don't the need to track the details about
   ///  how we parse this rule.
-  List<ParseTree> children;
+  List<ParseTree>? children;
 
   /// Get the initial/final token in this context.
   /// Note that the range from start to stop is inclusive, so for rules that do not consume anything
   /// (for example, zero length or error productions) this token may exceed stop.
-  Token start, stop;
+  Token? start, stop; // Todo: keep an eye on this late
```

Comment:
is this suppose to be nullable or `late`?

Reply:
In Java, looks like start and stop can be null
https://github.com/antlr/antlr4/blob/b5648f405b746d87ec559132ad1939fb6d402053/runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java#L306

Also it's a public attribute, and often not set through within the class, so I think it's better to leave it nullable (and keep the corresponding null checks).